### PR TITLE
Fix fixing ci

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

Fixes problem of Juju incompatibility in CI.

## Checklist

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I have bumped the version of any required library.

### CODE REVIEW

**Review by saltiyazan — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -81,13 +81,13 @@ jobs:
           /usr/bin/sudo --preserve-env=http_proxy,https_proxy,no_proxy pip3 install tox
 
           echo "# Install Juju"
-          /usr/bin/sudo snap install juju --classic --channel=latest/stable
+          /usr/bin/sudo snap install juju --classic --channel=3.1/stable
```
Comment (saltiyazan) [inline .github/workflows/main.yml]: can't we use the env variables here too?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -81,13 +81,13 @@ jobs:
...
 
           echo "# Install charmcraft"
           /usr/bin/sudo snap install charmcraft --classic --channel=latest/stable
 
           echo "# Install microk8s"
-          /usr/bin/sudo snap install microk8s --classic
+          /usr/bin/sudo snap install microk8s --channel=1.26-strict/stable
```
Comment (sanchezfdezjavier) [inline .github/workflows/main.yml]: Same as Yazan said above here?

File: .github/workflows/main.yml (hunk 1 above)
Comment (Gmerold) [reply .github/workflows/main.yml]: Guess I just missed it.

**Review by saltiyazan — APPROVED** [review]

File: .github/workflows/nginx.yml [hunk 3]
```diff
@@ -22,76 +22,9 @@ jobs:
...
-        uses: actions/checkout@v2
-      - name: Setup operator environment
-        run: |
-          echo "# Install core snap"
-          /usr/bin/sudo snap install core
-
-          echo "# Install LXD"
-          /usr/bin/sudo apt-get remove -qy lxd lxd-client || true
-          /usr/bin/sudo snap install lxd
-
```
Comment (saltiyazan) [inline .github/workflows/nginx.yml:38]: Installing lxd was not extracted to the new `integration-tests-self-hosted.yml` file.
Is that intended?

Comment (saltiyazan) [reply .github/workflows/nginx.yml:38]: Same for install core.

File: .github/workflows/integration-tests-self-hosted.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+on:
+  workflow_call:
+    inputs:
+      charm_dir:
+        required: true
+        type: string
+
+jobs:
+  orc8r-accessd-integration-test:
```
Comment (saltiyazan) [inline .github/workflows/integration-tests-self-hosted.yml]:
```suggestion
  integration-test:
```

Comment (saltiyazan) [reply .github/workflows/integration-tests-self-hosted.yml]: editted.

File: .github/workflows/nginx.yml (hunk 3 above)
Comment (Gmerold) [reply .github/workflows/nginx.yml:38]: Yes, we don't need it. We never did actually.

**Pull request merged**